src/facebook/facebookdatatypesyncadaptor.cpp

```diff
@@ -220,7 +220,7 @@ void FacebookDataTypeSyncAdaptor::signOnError(const SignOn::Error &error)
 
     // if the error is because credentials have expired, we
     // set the CredentialsNeedUpdate key.
-    if (error.type() == SignOn::AuthSession::UserInteractionError) {
+    if (error.type() == SignOn::Error::UserInteraction) {
         setCredentialsNeedUpdate(account);
```

Comment:
The AuthSession::ErrorType values are simply deprecated (and not used within signond or passed back to clients, from what I can grep).  If the value of AuthSession::UserInteractionError does not match the value of Error::UserInteraction then we would have been treating some OTHER error (eg, maybe Error::Network) as if it were a UserInteraction error.